account_journal_display_type/models/account_journal.py

```diff
@@ -0,0 +1,80 @@
+from odoo import api, fields, models
+
+
+class AccountJournal(models.Model):
+    _inherit = "account.journal"
+
+    display_type = fields.Selection(
+        [
+            ("sale", "Sales"),
+            ("purchase", "Purchase"),
+            ("cash", "Cash"),
+            ("bank", "Bank"),
+            ("payment", "Payment"),
+            ("general", "Miscellaneous"),
+        ],
+        required=True,
+        help="Select 'Sale' for customer invoices journals.\n"
+        "Select 'Purchase' for vendor bills journals.\n"
+        "Select 'Cash' or 'Bank' for journals that are used in customer or vendor payments.\n"
+        "Select 'General' for miscellaneous operations journals.",
+    )
+    type = fields.Selection(compute="_compute_type", store=True)
```

Comment:
type is python builtin and then shouldn't be used as a variable : you may use `type_` or `kind`

Reply:
the type is define in odoo core, the goal is to redefine this field to handle "payment" journal as type "bank" without useless account reference (default_account_id will have the same value as payment_debit_account_id). 

If display type is payment, payment_debit_account_id and payment_credit_account_id are sufficient required fields.
If display type is cash, the only field we keep is default_account_id. the payment config tabs is useless for a cash journal. (in this case payment_debit_account_id and payment_credit_account_id are set with value of default_account_id)  

https://github.com/OCA/OCB/blob/14.0/addons/account/models/account_account.py#L13

Reply:
Oops you're right, sorry